autodial-worker/src/lib/getRollingAverageRate.ts

```
@@ -0,0 +1,47 @@
+  const valBefore = df(preTime);
+  const vdelta = valNow - valBefore;
+  const divisor = weekendsOff ? window * 5 / 7 : window;
+
+  return vdelta / divisor;
+}
+
+function autoSum(data: Datapoint[]): Datapoint[] {
+  return data.reduce((prev: Datapoint[], p) => {
+    const last = prev[prev.length - 1];
+    const sum = last ? last.value + p.value : p.value;
+    return [...prev, {...p, value: sum}];
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)